src/server/cluster/cluster_family.cc

```diff
@@ -712,19 +713,38 @@ void ClusterFamily::DflyClusterMigrationFinalize(CmdArgList args, ConnectionCont
     return cntx->SendError("Migration process is not in STABLE_SYNC state");
   }
 
-  shard_set->pool()->AwaitFiberOnAll([migration](auto*) {
-    if (const auto* shard = EngineShard::tlocal(); shard)
+  // TODO implement blocking on migrated slots only
+  [[maybe_unused]] const auto deleted_slots = ToSlotSet(migration->GetSlotRange());
+
+  std::atomic_bool is_block_active = true;
+  auto is_pause_in_progress = [&is_block_active] { return is_block_active.load(); };
```

Comment:
You can't keep local references to something that you hand off, imagine:
1. polling fiber checks state -> false, so it sleeps
2. you finish migration -> scope aborts
3. polling fiber wakes up and reads from in invalid scope

Reply:
Also you don't need it to be atomic because we assume both fibers stay on the same thread, if we don't we have more problems than just an atomic 🙂

Reply:
What comes to mind is either retuning fiber ref from Pause or using shared_ptr<bool> for polling, maybe there are more options